packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2676,4 +2682,25 @@ export class TransactionController extends BaseControllerV1<
       log('Error while updating post transaction balance', error);
     }
   }
+
+  private async publishTransactionWithValidation(
```

Comment:
We've a pending ticket to remove the duplication between cancel and speed up, so I appreciate this method!

Perhaps a more explicit name would be `publishTransactionForRetry` so we know it's specific to speed up and cancel?

Reply:
Done in https://github.com/MetaMask/core/pull/3800/commits/27847911055229ab5ae0e3f748a2b0c741ed46e4